v6.0.0 to v6.0.1 introduces TypeScript type incompatibilities

## Description

In this change: https://github.com/Netflix/pollyjs/compare/v6.0.0...v6.0.1

The `NodeHttpAdapter` and `FSPersister` types are no longer compatible.

```ts
import NodeHttpAdapter from '@pollyjs/adapter-node-http';
import FSPersister from '@pollyjs/persister-fs';

  setupPolly({
    adapters: [NodeHttpAdapter],
    persister: FSPersister,
   ...
  });
```

### Error Message & Stack Trace

```
Type 'typeof NodeHttpAdapter' is not assignable to type 'string | typeof Adapter'.
  Type 'typeof NodeHttpAdapter' is not assignable to type 'typeof Adapter'.
    Construct signature return types 'NodeHttpAdapter' and 'Adapter<TOptions, TRequest>' are incompatible.
      The types of 'defaultOptions' are incompatible between these types.
        Type '{}' is not assignable to type 'TOptions'.
          '{}' is assignable to the constraint of type 'TOptions', but 'TOptions' could be instantiated with a different subtype of constraint '{}'.ts(2322)
```

## Environment

```
Node.js v14.18.1
linux 5.15.5-76051505-generic
npm: 6.14.15
```

Thanks for looking at this. Moved on to some new errors:

```shell
node_modules/@pollyjs/core/types.d.ts:108:3 - error TS7010: 'constructor', which lacks return-type annotation, implicitly has an 'any' return type.

108   constructor(
      ~~~~~~~~~~~~
109     polly: Polly,
    ~~~~~~~~~~~~~~~~~
... 
116     }
    ~~~~~
117   );
    ~~~~

node_modules/@pollyjs/persister/types.d.ts:2:24 - error TS7016: Could not find a declaration file for module 'set-cookie-parser'. '/<my-repo-redacated>/node_modules/set-cookie-parser/lib/set-cookie.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/set-cookie-parser` if it exists or add a new declaration (.d.ts) file containing `declare module 'set-cookie-parser';`

2 import { Cookie } from 'set-cookie-parser';
                         ~~~~~~~~~~~~~~~~~~~

Found 2 errors.
```

--- a/packages/@pollyjs/core/types.d.ts
+++ b/packages/@pollyjs/core/types.d.ts
@@ -105,16 +105,6 @@ export type RequestArguments = { [key: string]: any };
 
 export interface Request<TArguments extends RequestArguments = {}>
   extends HTTPBase {
-  constructor(
-    polly: Polly,
-    request: {
-      url: string;
-      method: string;
-      headers: Headers;
-      body?: string;
-      requestArguments: TArguments;
-    }
-  );
   method: string;
   url: string;
   readonly absoluteUrl: string;

--- a/packages/@pollyjs/persister/package.json
+++ b/packages/@pollyjs/persister/package.json
@@ -40,6 +40,7 @@
   "license": "Apache-2.0",
   "dependencies": {
     "@pollyjs/utils": "^6.0.1",
+    "@types/set-cookie-parser": "^2.4.1",
     "bowser": "^2.4.0",
     "fast-json-stable-stringify": "^2.1.0",
     "lodash-es": "^4.17.21",

